### PR TITLE
ISPN-16309 Throw a descriptive exception if user properties are malfo…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-16309

### CODE REVIEW

**Review by jabolina — APPROVED** [review]
Is it OK to the exception to show the username of an existing user?

File: server/runtime/src/main/java/org/infinispan/server/security/realm/EncryptedPropertiesSecurityRealm.java [hunk 1]
```diff
@@ -271,6 +271,9 @@ public void load(InputStream usersStream, InputStream groupsStream) throws IOExc
                               String[] passwords = builder.toString().trim().split(";");
                               for (String password : passwords) {
                                  int colon = password.indexOf(':');
+                                 if (colon < 0) {
+                                    throw log.malformedUserProperties(username);
```
Comment (wburns) [inline server/runtime/src/main/java/org/infinispan/server/security/realm/EncryptedPropertiesSecurityRealm.java:275]: I agree with Jose, maybe instead we can just keep track of what line each username is on and instead state there is a malformed user name at line number X? I am thinking if a user may have access to the server log but not write/read access to the server configuration itself.

**Comment by tristantarrant** [comment]
I have changed this to just mention the line number

**Comment by jabolina** [comment]
Thanks, @tristantarrant!

**Pull request merged**